WalletWasabi.Gui/Controls/WalletExplorer/CoinListViewModel.cs

```diff
@@ -314,6 +325,28 @@ public CoinListViewModel()
 
 			SortCommand = ReactiveCommand.Create(() => RefreshOrdering()).DisposeWith(Disposables);
 
+			SignMessage = ReactiveCommand.Create(() =>
+			{
+				OnEncryptionManager(EncryptionManagerViewModel.Tabs.Sign, SelectedCoin.Address);
+			}).DisposeWith(Disposables);
+
+			VerifyMessage = ReactiveCommand.Create(() =>
+			{
+				OnEncryptionManager(EncryptionManagerViewModel.Tabs.Verify, SelectedCoin.Address);
+			}).DisposeWith(Disposables);
+
+			EncryptMessage = ReactiveCommand.Create(() =>
+			{
+				var res = Global.WalletService.KeyManager.GetKeyForScriptPubKey(SelectedCoin.Model.ScriptPubKey)?.PubKey?.ToHex();
```

Comment:
C1: Why is this `)?.PubKey?.ToHex()`? I mean, is it possible to have a coin which private key cannot be found? I am not sure but I think it is not possible.
C2: I think if [this PR](https://github.com/zkSNACKs/WalletWasabi/pull/1145) is merged first then you could take advantage of the fact that it added the `Pubkey` property to the `SmartCoin` so, this line could be removed or simply replaced by:
 ```
var res = SelectedCoin.Model.Pubkey;
```
C3: I think `res` should be renamed by `publicKey`, `publicKeyHex` or similar.

Reply:
fixed!